RV64I.core_desc

```diff
@@ -9,146 +9,150 @@ InstructionSet RV64I extends RV32I {
-                    X[rd % RFS] = (unsigned<64>)(signed)sh_val;
+            behavior: if(rd >=RFS || rs1 >= RFS) raise(0, 2); else {
+                if (rd != 0) {
+                    unsigned<32> sh_val = ((unsigned<32>)X[rs1]) >> shamt;
+                    X[rd] = (signed<XLEN>)(signed)sh_val;
                 }
             }
         }
 
         SRAIW {
             encoding: 7'b0100000 :: shamt[4:0] :: rs1[4:0] :: 3'b101 :: rd[4:0] :: 7'b0011011;
             assembly:"{name(rd)}, {name(rs1)}, {shamt}";
-            behavior: {
-                if ((rd % RFS) != 0) {
-                    signed<32> sh_val = ((signed<32>)X[rs1 % RFS]) >> shamt;
-                    X[rd % RFS] = (unsigned<64>)sh_val;
+            behavior: if(rd >=RFS || rs1 >= RFS) raise(0, 2); else {
+                if (rd != 0) {
+                    signed<32> sh_val = ((signed<32>)X[rs1]) >> shamt;
+                    X[rd] = (signed<XLEN>)sh_val;
                 }
             }
         }
 
         ADDW {
             encoding: 7'b0000000 :: rs2[4:0] :: rs1[4:0] :: 3'b000 :: rd[4:0] :: 7'b0111011;
-            behavior: {
-                if ((rd % RFS) != 0) {
-                    signed<32> res = (signed<32>)X[rs1 % RFS] + (signed<32>)X[rs2 % RFS];
-                    X[rd % RFS] = (signed<64>)res;
+            behavior: if(rd >=RFS || rs1 >= RFS || rs2 >= RFS) raise(0, 2); else {
+                if (rd != 0) {
+                    signed<32> res = (signed<32>)X[rs1] + (signed<32>)X[rs2];
+                    X[rd] = (signed<XLEN>)res;
                 }
             }
         }
 
         SUBW {
             encoding: 7'b0100000 :: rs2[4:0] :: rs1[4:0] :: 3'b000 :: rd[4:0] :: 7'b0111011;
-            behavior: {
-                if ((rd % RFS) != 0) {
-                    signed<32> res = (signed<32>)X[rs1 % RFS] - (signed<32>)X[rs2 % RFS];
-                    X[rd % RFS] = (signed<64>)res;
+            behavior: if(rd >=RFS || rs1 >= RFS || rs2 >= RFS) raise(0, 2); else {
+                if (rd != 0) {
+                    signed<32> res = (signed<32>)X[rs1] - (signed<32>)X[rs2];
+                    X[rd] = (signed<XLEN>)res;
                 }
             }
         }
 
         SLLW {
             encoding: 7'b0000000 :: rs2[4:0] :: rs1[4:0] :: 3'b001 :: rd[4:0] :: 7'b0111011;
             assembly:"{name(rd)}, {name(rs1)}, {name(rs2)}";
-            behavior: {
-                if ((rd % RFS) != 0) {
-                    unsigned<32> count = (unsigned)X[rs2 % RFS] & 0x1f;
-                    unsigned<32> sh_val = ((unsigned<32>)X[rs1 % RFS]) << count;
-                    X[rd % RFS] = (unsigned<64>)(signed)sh_val;
+            behavior: if(rd >=RFS || rs1 >= RFS || rs2 >= RFS) raise(0, 2); else {
+                if (rd != 0) {
+                    unsigned<32> count = (unsigned)X[rs2] & 0x1f;
+                    unsigned<32> sh_val = ((unsigned<32>)X[rs1]) << count;
+                    X[rd] = (signed<XLEN>)(signed)sh_val;
```

Comment:
Unsure about this change here...

Reply:
this will still work but should really be `X[rd] = (unsigned<XLEN>)(signed)sh_val;`, assigning a signed RHS to an unsigned LHS of the same size is considered loss of precision and therefore erroneous behavior per the CoreDSL 2 spec.